src/redisearch_rs/c_entrypoint/types_ffi/src/lib.rs

```diff
@@ -78,9 +78,12 @@ pub unsafe extern "C" fn IndexResult_SetNumValue(result: *mut RSIndexResult, val
 /// The following invariant must be upheld when calling this function:
 /// - `result` must point to a valid `RSIndexResult` and cannot be NULL.
 #[unsafe(no_mangle)]
-pub unsafe extern "C" fn IndexResult_TermRef(
-    result: *const RSIndexResult<'_>,
-) -> Option<&RSTermRecord<'_>> {
+pub unsafe extern "C" fn IndexResult_TermRef<'index, 'aggregate_children>(
+    result: *const RSIndexResult<'index, 'aggregate_children>,
+) -> Option<&'index RSTermRecord<'index>>
```

Comment:
This isn't quite correct. The lifetime of the returned reference is tied to the lifetime of `result`.
Morally, this takes as input a `&'a RSIndexResult<'index, 'aggregate_children>` and returns a `&'a RSTermRecord<'index>`, where `'index` must outlive `'a`.

I think it'd be easier, at this stage, to return a `*const RSTermRecord<'index>` or `Option<NonNull<RSTermRecord<'index>>>`.

Reply:
If we choose to return a raw pointer,

```rust
#[unsafe(no_mangle)]
pub unsafe extern "C" fn IndexResult_TermRef<'result, 'index, 'aggregate_children>(
    result: *const RSIndexResult<'index, 'aggregate_children>,
) -> *const RSTermRecord<'index>
where
    'aggregate_children: 'result,
    'index: 'result,
{
    debug_assert!(!result.is_null(), "result must not be null");

    // SAFETY: Caller is to ensure that the pointer `result` is a valid, non-null pointer to
    // an `RSIndexResult`.
    let result: &'result RSIndexResult<'index, 'aggregate_children> = unsafe { &*result };

    result
        .as_term()
        .map_or(std::ptr::null(), move |r| r as *const _)
}
```